rakelib/license.rake

```diff
@@ -0,0 +1,45 @@
+# we need to call exit explicity  in order to set the proper exit code, otherwise
+# most common CI systems can not know whats up with this tests.
+require_relative "default_plugins"
+
+namespace "license" do
+
+  desc "run core specs"
+  #task "core" =>  ["test:setup"] do
+  task "core" do
+    require 'yaml'
+    generate_license_information
+  end
+
+end
+
+def all_installed_gems
+  Gem::Specification.all = nil
+  all = Gem::Specification
+  Gem::Specification.reset
+  all
+end
+
+def generate_license_information
+  licenses = []
+  all_installed_gems.select {|y| y.gem_dir.include?('vendor') }.sort {|v, u| v.name  <=> u.name }.each do |x|
```

Comment:
Pretty clear to me. but I wrote it.  Putting into function or breaking into multiline is fine